src/main/java/bio/terra/janitor/service/janitor/TrackedResourceService.java

```diff
@@ -116,60 +108,47 @@ private TrackedResourceId createResourceAndUpdateDuplicates(
     TrackedResourceId trackedResourceId = TrackedResourceId.create(uuid);
     Optional<TrackedResourceAndLabels> resourceAndLabels =
         janitorDao.retrieveResourceAndLabels(trackedResourceId);
-    return resourceAndLabels.map(JanitorService::createInfo);
+    return resourceAndLabels.map(TrackedResourceService::createInfo);
   }
 
   /** Retrieves the resources with the {@link CloudResourceUid}. */
-  public TrackedResourceInfoList getResources(
-      CloudResourceUid cloudResourceUid, AuthenticatedUserRequest userReq) {
-    iamService.requireAdminUser(userReq);
+  public TrackedResourceInfoList getResources(CloudResourceUid cloudResourceUid) {
     List<TrackedResourceAndLabels> resourcesWithLabels =
         janitorDao.retrieveResourcesWith(cloudResourceUid);
     TrackedResourceInfoList resourceList = new TrackedResourceInfoList();
     resourcesWithLabels.stream()
-        .map(JanitorService::createInfo)
+        .map(TrackedResourceService::createInfo)
         .forEach(resourceList::addResourcesItem);
     return resourceList;
   }
 
+  /** Updates the READY resource state to ABANDONED. */
+  public void abandonResource(CloudResourceUid cloudResourceUid) {
+    List<TrackedResource> resources =
+        getResourceWithState(
+            cloudResourceUid, TrackedResourceState.READY, TrackedResourceState.CLEANING);
+
+    if (resources.size() > 1) {
+      throw new InternalServerErrorException(
+          "More than one READY or CLEANING state resources are found.");
```

Comment:
Hmmm, maybe we should log an error but still abandon all of the resources anyway. If we're in a bad state where there are multiple active resources, abandoning all of them seems reasonable. We do want to indicate with a log error that it's a bad state. WDYT?

Reply:
Done